lib/hashie/mash.rb

```diff
@@ -350,8 +350,11 @@ def log_built_in_message(method_key)
     end
 
     def log_collision?(method_key)
+      @ignore_suffix = true
       respond_to?(method_key) && !self.class.disable_warnings? &&
         !(regular_key?(method_key) || regular_key?(method_key.to_s))
+    ensure
+      @ignore_suffix = false
```

Comment:
Without looking at the PR explanation this code rescues anything and turns it into `@ignore_suffix`. At the very least you should turn this into some explicit check and not rely on the exception.

Reply:
This probably isn't the best way to handle ignoring the error, although I'm a little confused what you mean. The ensure block does not require an exception to execute and does not prevent an exception from being raised.

Reply:
Sorry I wasn't clear either. Yes, implement the check more explicitly instead of using `ensure`. It can be as dumb as a version check for Ruby 2.2, but frankly that all seems weird and I'd want to dig deeper to find what the real problem on that specific ruby version is maybe with the help of some Ruby people?

Is it true that you started on fixing (2) and then into (1)? Maybe alter this PR for that first that only fails on 2.2 and I can help with that?

Reply:
I don't think we aren't quite on the same page still. So, I cut a new PR to demonstrate the first issue (1) in order to hopefully make it clearer (https://github.com/intridea/hashie/pull/443)

This PR now only addresses the second issue (2).

The ensure is not meant for rescuing the NameError, it was to ensure the ```@ingnore_underscore``` instance variable was unset without messing with the returned result of ```log_collision?```.

Again, I don't love the use of the instance variable, I'm just not sure how else to alter the behavior of ```responds_to_missing?``` only within the context of the ```log_collision?``` check. I'm definitely open to suggestions.